### DISCUSSION TITLE
How to use kyuubi to get which partition of table does spark sql hit？

### DISCUSSION THREAD

**Discussion by madeirak** [apache/kyuubi#6738] [Q&A]
How to use kyuubi to get which partition of table does spark sql hit？
Similar to SQL Lineage Support, we can know which table does spark sql hit.
https://kyuubi.readthedocs.io/en/master/extensions/engines/spark/lineage.html

**Comment by pan3793** [comment]
You can write a custom Spark plugin to traverse the query plan and extract the table scan partition info.

**Reply by madeirak** [reply]
thx for reply, I want to use kyuubi custom eventhandler to trigger subsequent processing. I noticed Kyuubi is mainly Scala. Can custom eventhandler be recognized successfully when written in Java??

**Reply by pan3793** [reply]
Scala and Java should be interoperable, it's technically possible. I don't have much experience in invoking Scala from the Java code, so don't expect me to provide any specific details.
